airflow/executors/kubernetes_executor.py

```diff
@@ -287,15 +293,35 @@ def _make_kube_watcher(self) -> KubernetesJobWatcher:
         watcher.start()
         return watcher
 
-    def _health_check_kube_watcher(self):
-        if self.kube_watcher.is_alive():
-            self.log.debug("KubeJobWatcher alive, continuing")
-        else:
-            self.log.error(
-                "Error while health checking kube watcher process. Process died for unknown reasons"
+    def _make_kube_watchers(self) -> dict[str | None, KubernetesJobWatcher]:
+        watchers = {}
+        if self.kube_config.multi_namespace_mode:
+            namespaces_to_watch = (
+                self.kube_config.multi_namespace_mode_namespace_list
+                if self.kube_config.multi_namespace_mode_namespace_list
+                else [None]
             )
-            ResourceVersion().resource_version = "0"
-            self.kube_watcher = self._make_kube_watcher()
+        else:
+            namespaces_to_watch = [self.kube_config.kube_namespace]
+
+        for namespace in namespaces_to_watch:
+            watchers[namespace] = self._make_kube_watcher(namespace)
+        return watchers
+
+    def _health_check_kube_watchers(self):
+        for namespace, kube_watcher in self.kube_watchers.items():
+            if kube_watcher.is_alive():
+                self.log.debug("KubeJobWatcher for namespace %s alive, continuing", namespace)
+            else:
+                self.log.error(
+                    (
+                        "Error while health checking kube watcher process for namespace %s. "
+                        "Process died for unknown reasons"
+                    ),
+                    namespace,
+                )
+                ResourceVersion().resource_version[namespace] = "0"
+                self.kube_watchers[namespace] = self._make_kube_watcher(namespace)
 
     def run_next(self, next_job: KubernetesJobType) -> None:
```

Comment:
in this method construct_pod is called with `self.namespace`. does this make sense with multi-namespace mode?  how are additional namespaces used if pods are always created with the one namespace?

Reply:
@dstandish  , that's a good question. 

Actually along preparing this PR, I kept having this question: did this `multi_namespace_mode` feature ever really work for anyone? I also noticed a few similar issues, and I shared them at https://lists.apache.org/thread/nqnq2gbvtx6fyycsqhgjhlz1yotq0vhc

Reply:
It looks like maybe if you don't set `namespace` in airflow cfg then you may be able to set it with executor config

i don't know more than that

are you using it now?

Reply:
Our team are using KubernetesExecutor, and we do run it with with a multi namespace setting. But we couldn't use vanilla `multi_namespace_mode` as it is. That's exactly why I'm proposing the changes in this PR ;-)

Reply:
> It looks like maybe if you don't set namespace in airflow cfg then you may be able to set it with executor config

And yes, I believe this is the answer to the original question you shared above.

Reply:
cool